awscli/customizations/s3/s3handler.py

```diff
@@ -635,6 +645,23 @@ def _warn_glacier(self, fileinfo):
                 return True
         return False
 
+    def _format_fileinfo_src_dest(self, fileinfo):
+        """Returns formatted versions of a fileinfos source and destination."""
+        src = self._format_path(fileinfo.src, fileinfo.src_type)
+        dest = self._format_path(fileinfo.dest, fileinfo.dest_type)
+        return src, dest
+
+    def _format_path(self, path, path_type):
```

Comment:
So I like what you did by making this internal and then implementing the ``DeleteRequestSubmitter._format_path``, but I feel like you can leverage the fact that we have already resolved to what type of transfer we are about to do. So I can see the following happen:

* Make ``BaseTransferRequestSubmitter._format_path`` raise a ``NotImplementedError`` and have them implemented in the child classes.
* Create helper methods for specific types of formating (i.e. ``_format_local_path()`` and ``_format_s3_path()``
* Remove the ``if path == '-'`` check in the ``_format_local_path()`` and just have the request submitters for those stream just return ``-``.

While it is a bit more code, I think it is beneficial because we would be minimizing the coupling between the FileInfo's and RequestSubmitters. With having a general method that generates the formatted path, we have to rely on more of the values from the ``FileInfo`` such as ``path_type`` when we really do not need to because we already know the transfer type ahead of time. In the future, if we were to refactor ``FileInfo`` (as I think it currently does too much and too many abstractions are coupled to it), we will not have to worry about affecting this functionality.

Let me know if this make sense?